File: backend-ms1/src/main/java/ma/zs/stocky/service/facade/admin/produit/ProduitAdminService.java
package ma.zs.stocky.service.facade.admin.produit;

import java.util.List;
import ma.zs.stocky.bean.core.produit.Produit;
import ma.zs.stocky.dao.criteria.core.produit.ProduitCriteria;
import ma.zs.stocky.zynerator.service.IService;



public interface ProduitAdminService extends  IService<Produit,ProduitCriteria>  {

    List<Produit> findByCategorieProduitId(Long id);
    int deleteByCategorieProduitId(Long id);
    long countByCategorieProduitId(Long id);




}

File: frontend/src/app/shared/service/admin/produit/ProduitAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {ProduitDto} from 'src/app/shared/model/produit/Produit.model';
import {ProduitCriteria} from 'src/app/shared/criteria/produit/ProduitCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class ProduitAdminService extends AbstractService<ProduitDto, ProduitCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): ProduitDto {
        return new ProduitDto();
    }

    public constrcutCriteria(): ProduitCriteria {
        return new ProduitCriteria();
    }

    get API() {
        return environment.apiUrlStockyservice + 'admin/produit/';
    }
}
